trust_stores_observatory/store_fetcher/java_fetcher.py

```diff
@@ -18,170 +20,161 @@
-
-            trusted_certificates = RootRecordsValidator.validate_with_repository(
-                cert_repo, root_records)
-            blacklisted_certificates = RootRecordsValidator.validate_with_repository(
-                cert_repo, blacklisted_records)
-
-        return TrustStore(PlatformEnum.ORACLE_JAVA, version, url,
-                          datetime.utcnow().date(), trusted_certificates,
-                          blacklisted_certificates)
+            jre_temp_file.write(response.read())
+            jre_temp_file.close()
+            with JrePackage(jre_temp_file.name) as parsed_jre:
+                # Extract the data we need
+                version = parsed_jre.get_version()
+                blacklisted_file_content = parsed_jre.get_blacklisted_certs()
+                cacerts_key_store = jks.KeyStore.loads(parsed_jre.get_cacerts(), parsed_jre.get_cacerts_password())
+        finally:
+            os.remove(jre_temp_file.name)
+
+        # Process the data extracted from the JRE
+        # Trusted CA certs
+        scraped_trusted_records = self._extract_trusted_root_records(cacerts_key_store, should_update_repo, cert_repo)
+        trusted_records = RootRecordsValidator.validate_with_repository(cert_repo, scraped_trusted_records)
+
+        # Blacklisted CA certs - will fail if a blacklisted cert is not already available in the local repo
+        scraped_blacklisted_records = self._extract_blacklisted_root_records(blacklisted_file_content)
+        blacklisted_records = RootRecordsValidator.validate_with_repository(cert_repo, scraped_blacklisted_records)
+
+        return TrustStore(
+            PlatformEnum.ORACLE_JAVA,
+            version,
+            final_url,
+            datetime.utcnow().date(),
+            trusted_records,
+            blacklisted_records
+        )
 
     @staticmethod
-    def _parse_root_records(key_store: jks.KeyStore, should_update_repo: bool,
-                            cert_repo: RootCertificatesRepository
-                            ) -> List[ScrapedRootCertificateRecord]:
+    def _extract_trusted_root_records(
+            key_store: jks.KeyStore,
+            should_update_repo: bool,
+            cert_repo: RootCertificatesRepository
+    ) -> List[ScrapedRootCertificateRecord]:
         root_records = []
         for alias, item in key_store.certs.items():
-            cert = load_der_x509_certificate(item.cert, default_backend())
-
+            parsed_cert = load_der_x509_certificate(item.cert, default_backend())
             if should_update_repo:
-                cert_repo.store_certificate(cert)
+                cert_repo.store_certificate(parsed_cert)
 
-            fingerprint = cert.fingerprint(hashes.SHA256())
-            subject_name = ''
-
-            try:
-                subject_name = cert.subject.get_attributes_for_oid(
-                    NameOID.COMMON_NAME)[0].value
-            except Exception:
```

Comment:
This logic was already available in `CertificateUtils.get_canonical_subject_name()`.